ddtrace/profiling/collector/_lock.py

```diff
@@ -233,6 +235,47 @@ def __enter__(self, *args, **kwargs):
     def __exit__(self, *args, **kwargs):
         self._release(self.__wrapped__.__exit__, *args, **kwargs)
 
+    # Get lock acquire/release call location and variable name the lock is assigned to
+    def _get_lock_call_loc_with_name(self) -> typing.Optional[str]:
+        try:
+            # We expect the call stack to be like this:
+            # 0: this
+            # 1: _acquire/_release
+            # 2: acquire/release (or __enter__/__exit__)
+            # 3: caller frame
+            # And we expect addtioinal frame if WRAPT_C_EXT is False
+            frame = sys._getframe(3 if WRAPT_C_EXT else 4)
+            code = frame.f_code
+            call_loc = "%s:%d" % (os.path.basename(code.co_filename), frame.f_lineno)
+
+            var_name = None
+
+            # Search for global variables
+            for name, value in frame.f_globals.items():
+                if value == self:
+                    var_name = name
+                    break
+                for attribute in dir(value):
+                    if not attribute.startswith("__") and getattr(value, attribute) == self:
+                        var_name = attribute
+                        break
+            # Search for local variables
+            for name, value in frame.f_locals.items():
+                if value == self:
+                    var_name = name
+                    break
+                for attribute in dir(value):
+                    if not attribute.startswith("__") and getattr(value, attribute) == self:
+                        var_name = attribute
+                        break
+
+            if var_name:
+                call_loc += ":%s" % var_name
```

Comment:
If we're here, then we tried to find a variable name and we weren't able to.  We know that users are going to hit this condition and they're going to wonder why their lock is unnamed.

Users are generally not going to read the documentation for every feature.  They're going to interact with these objects in the UX first and foremost, so the names of things tend to reveal not only the context of objects within their application, but also the state of the _instrumentation_ when that context was gathered.  I propose adding some kind of default name indicating that the lookup for the symbol failed.

I wouldn't be totally heartbroken if you decided to keep it blank, since I personally interpret what you have to mean "There's no more info, go read the docs if this is a bummer for you."

Reply:
Maybe that's a TODO in UI? In UI, we could check whether we have variable name by checking that the lock name is of the form `%s:%d:%s`or what not. And then show a button or something like "Can't find variable name for the lock?" and link to the docs?